csi/exporter.py

```diff
@@ -1,129 +1,142 @@
-                used = total - free
-                capacity_bytes.add_metric(labels, total)
-                capacity_free_bytes.add_metric(labels, free)
-                capacity_used_bytes.add_metric(labels, used)
-
-                # Inodes
-                total = stat.f_files
-                free = stat.f_favail
-                used = total - free
-                inodes_count.add_metric(labels, total)
-                inodes_free_count.add_metric(labels, free)
-                inodes_used_count.add_metric(labels, used)
-
-                # Gathers capacity metrics for each subvol
-                for pvc in yield_pvc_from_mntdir(os.path.join(pth, "info")):
-                    # pvc[0]: name
-                    # pvc[1]: size
-                    # pvc[2]: path prefix
-                    pvcpath = os.path.join(pth, pvc[2])
-                    pvcname = pvc[0]
-                    pvclabels = labels + [pvcname]
-
-                    stat = os.statvfs(pvcpath)
-
-                    # Capacity
-                    total = stat.f_bsize * stat.f_blocks
-                    free = stat.f_bsize * stat.f_bavail
-                    used = total - free
-                    pv_capacity_bytes.add_metric(pvclabels, total)
-                    pv_capacity_free_bytes.add_metric(pvclabels, free)
-                    pv_capacity_used_bytes.add_metric(pvclabels, used)
-
-        yield capacity_bytes
-        yield capacity_free_bytes
-        yield capacity_used_bytes
-        yield inodes_count
-        yield inodes_free_count
-        yield inodes_used_count
-        yield pv_capacity_bytes
-        yield pv_capacity_used_bytes
-        yield pv_capacity_free_bytes
-
-
-REGISTRY.register(CsiMetricsCollector())
+app = FastAPI()
+
+@app.get("/_api/metrics")
+def metrics():
+    """
+    Gathers storage and pvcs metrics.
+    Starts process by exposing the data collected in port 8050 at '/_api/metrics'.
+    """
+
+    data = {
+        "pod": {},
+        "storages": []
+    }
+
+    pod_name = ""
+    memory_usage_in_bytes = 0
+    cpu_usage_in_nanoseconds = 0
+
+    pod_name_path = '/etc/hostname'
+    with open(pod_name_path, 'r') as pod_fd:
+        pod_name = pod_fd.read().strip()
+
+    memory_usage_file_path = '/sys/fs/cgroup/memory/memory.usage_in_bytes'
+    with open(memory_usage_file_path, 'r') as memory_fd:
+        memory_usage_in_bytes = int(memory_fd.read().strip())
+
+    cpu_usage_file_path = '/sys/fs/cgroup/cpu/cpuacct.usage'
+    with open(cpu_usage_file_path, 'r') as cpu_fd:
+        cpu_usage_in_nanoseconds = int(cpu_fd.read().strip())
+
+    data["pod"] = {
+        "pod_name": pod_name,
+        "memory_usage_in_bytes": memory_usage_in_bytes,
+        "cpu_usage_in_nanoseconds": cpu_usage_in_nanoseconds
+    }
+
+    # Handle condition for no storage & PVC,
+    # sometimes storage name is not shown at /mnt until server is mounted.
+    if len(os.listdir(HOSTVOL_MOUNTDIR)) == 0:
+        logging.error(logf(
```

Comment:
Remove error, change it as debug message. Since this is a valid usecase till a Storage pool is added/mounted.